lib/galaxy/jobs/__init__.py

```diff
@@ -1809,27 +1809,11 @@ def requires_setting_metadata( self ):
             return self.tool.requires_setting_metadata
         return False
 
-    def _report_error_to_sentry( self ):
+    def _report_error( self ):
         job = self.get_job()
         tool = self.app.toolbox.get_tool(job.tool_id, tool_version=job.tool_version) or None
-        if self.app.sentry_client and job.state == job.states.ERROR:
-            self.app.sentry_client.capture(
-                'raven.events.Message',
-                message="Galaxy Job Error: %s  v.%s" % (job.tool_id, job.tool_version),
-                extra={
-                    'info' : job.info,
-                    'id' : job.id,
-                    'command_line' : job.command_line,
-                    'stderr' : job.stderr,
-                    'traceback': job.traceback,
-                    'exit_code': job.exit_code,
-                    'stdout': job.stdout,
-                    'handler': job.handler,
-                    'user': self.user,
-                    'tool_version': job.tool_version,
-                    'tool_xml': tool.config_file if tool else None
-                }
-            )
+        for dataset in job.output_datasets:
+            self.app.error_reports.default_error_sink.submit_report(dataset, job, tool, user_submission=False)
```

Comment:
The job instrumenters you copied this from are per-job-destination. So different job destinations can have different job metrics collected. The default applies to any that aren't specified on a per destination basis - which is likely every destination ever since this wasn't really documented. I'd probably argue the error reporting should be overridable on a per destination basis for the same reason - but since I didn't document it replicating the full complexity of job metrics is not something I'd really require from you.

But to answer the question about the odd line:

```
    self.error_sinks = collections.defaultdict(lambda: self.default_error_sink)
```

It is meant to be used like:

```
    self.app.error_reports.error_sinks[job.destination_id].submit_report(dataset, job, tool, user_submission=False)
```

In this case, the lambda is just telling the ``defaultdict`` which set of error sink plugins to use if no per-destination one has been explicitly set. I'd make that change just to bring it inline with job metrics and because you have the job available right there - it allows us to grow in that direction in the future (per-destination error reporting) - even if you don't add the job conf hooks now.

Ideally everything is overridable at the destination this way. The nature of job destinations is also such that this means it is overridable from dynamic job destinations and on a per-tool basis for instance also.